packages/node/src/indexer/dictionary.service.ts

```diff
@@ -25,20 +22,29 @@ export class DictionaryService
     super(project.network.dictionary, project.network.chainId, nodeConfig);
   }
 
-  async getEvmChainId(): Promise<string> {
-    const query = `query{chainAlias(id: "evmChainId"){value}}`;
+  async init(): Promise<void> {
+    /*Some dictionarys for EVM are built with other SDKs as they are chains with an EVM runtime
+     * we maintain a list of aliases so we can map the evmChainId to the genesis hash of the other SDKs
+     * e.g moonbeam is built with Substrate SDK but can be used as an EVM dictionary
+     */
+    const chainAliases = await this.getEvmChainId();
+    const chainAlias = chainAliases[this.chainId];
 
-    try {
-      const resp = await timeout(
-        this.client.query({
-          query: gql(query),
-        }),
-        this.nodeConfig.dictionaryTimeout,
-      );
-      return resp.data.chainAlias.value;
-    } catch (e) {
-      logger.debug(`Dictionary doesn't have an evmChainId set`);
-      return undefined;
+    if (chainAlias) {
+      // Cast as any to work around read only
+      (this.chainId as any) = chainAlias;
     }
```

Comment:
shall we give any warnings if `chainAlias` got undefined?

Reply:
No because its quite common for there to not be chain aliases